lib/benchmark/ips/quick.rb

```diff
@@ -0,0 +1,32 @@
+module Benchmark
+  module IPS
+    # Quickly compare multiple methods on the same method.
+    # @param obj The object to call the methods on
+    # @param methods The list of methods to call as Symbols
+    #
+    # Keyword arguments are detected as well, with the following
+    # available:
+    #
+    # @param warmup How many seconds to warm up the benchmark process
+    # @param time How many seconds to benchmark each method
+    def quick_compare(obj, *methods)
+      if methods.last.kind_of? Hash
+        opts = methods.pop
+      else
+        opts = {}
+      end
+
+      Benchmark.ips do |x|
+        x.compare!
+        x.warmup = opts[:warmup] if opts[:warmup]
+        x.time = opts[:time] if opts[:time]
+
+        methods.each do |name|
+          x.report(name) do |x|
+            x.times { obj.__send__ name }
+          end
```

Comment:
```suggestion
          block = eval "proc { obj.#{name} }"
          x.report(name, &block)
```

This avoids the `__send__` overhead in an easy way and it also reuses the existing block form optimization to use a while loop (instead of times which needs another block call per iteration).

Reply:
Man... you're really focused on that send... Again: Overhead doesn't matter as long as it is evenly/fairly applied.

I was using the block arg form of report because my understanding is that you get better resolution. I assumed that `x` was tuned based on the warmup numbers. Is this not the case? My benchmark benchmarks show a 20% increase in performance using `x` + `x.times`.

Reply:
> Again: Overhead doesn't matter as long as it is evenly/fairly applied.

It does. Otherwise some differences cannot be measured with a big enough overhead.
A benchmarking harness should always aim to have minimal overhead.